app/src/main/java/com/epfl/drawyourpath/database/Database.kt

```diff
@@ -148,4 +166,89 @@ abstract class Database {
+     */
+    abstract fun getChatPreview(conversationId: String): CompletableFuture<ChatPreview>
+
+    /**
+     * Function used to modify the title of a given conversation with his conversationId
+     * @param conversationId that we want to modify the name
+     * @param newTitle new title of the conversation
+     * @return a future that indicate if the name of the conversation was correctly modify.
+     */
+    abstract fun setChatTitle(conversationId: String, newTitle: String): CompletableFuture<Unit>
+
+    /**
+     * Function used to get the members list of a given conversation with his conversationId
+     * @param conversationId that we want to retrieve the members list
+     * @return a future that contains the members list (a list of their userId) of the conversation
+     */
+    abstract fun getChatMemberList(conversationId: String): CompletableFuture<List<String>>
+
+    /**
+     * Function used to add a member in a given conversation with his conversation Id
+     * @param userId userId of the member that we want to add to the conversation
+     * @param conversationId of the conversation
+     * @return a future that indicate if the new member was correctly added
+     */
+    abstract fun addChatMember(userId: String, conversationId: String): CompletableFuture<Unit>
+
+    /**
+     * Function used to remove a member in a given conversation with his conversation Id
+     * @param userId userId of the member that we want to remove to the conversation
+     * @param conversationId of the conversation
+     * @return a future that indicate if the member was correctly removed
+     */
+    abstract fun removeChatMember(userId: String, conversationId: String): CompletableFuture<Unit>
+
+    /**
+     * Function to get the chat messages of a given conversation with his conversation Id
+     * @param conversationId of the conversation
+     * @return a future that contains the messages of the given conversation
+     */
+    abstract fun getChatMessages(conversationId: String): CompletableFuture<List<Message>>
+
+    /**
+     * Function used to add a message to the messages list of a given conversation with his conversationId
+     * @param conversationId of the conversation
+     * @param message that we want to add to the conversation
+     * @return a future that indicated if the message was correctly added to the database
+     */
+    abstract fun addChatMessage(conversationId: String, message: Message): CompletableFuture<Unit>
+
+    /**
+     * Function used to remove a message(with a given timestamp) from a conversation with his id
+     * @param conversationId of the conversation
+     * @param timestamp of the message that we want to removed(in epoch seconds)
+     * @return a future that indicated if the message was correctly deleted
+     */
+    abstract fun removeChatMessage(conversationId: String, timestamp: Long): CompletableFuture<Unit>
```

Comment:
Suggestion: Perhaps it would make the code more readable if timestamp was renames to message_id. I thinnk even if we use the timestamp as the id it would make the code easier to understand if its named id.